src/RollbarClientSubmitter.ts

```diff
@@ -12,31 +21,41 @@ const configurationDefaults = {
   setContext: () => window.location.href,
 };
 const configurationOptionsRequired = ['accessToken', 'environment'];
-const libraryName = process.env.npm_package_name;
-const libraryVersion = process.env.npm_package_version;
+const libraryName = process.env['npm_package_name'];
+const libraryVersion = process.env['npm_package_version'];
 
 // Local Functions
-function buildEnumerableObject(targetObject) {
-  const enumerableObject = {};
+function buildEnumerableObject(targetObject: Error) {
+  const enumerableObject: Record<string, string | Error> = {};
   for (const key of Object.getOwnPropertyNames(targetObject)) {
-    enumerableObject[key] = targetObject[key];
+    enumerableObject[key] = (targetObject as Required<Error>)[key as keyof Error];
   }
   return enumerableObject;
 }
 
-function buildObjectDeepSorted(targetValue) {
-  if (targetValue === null || typeof targetValue !== 'object' || Array.isArray(targetValue)) {
-    return targetValue;
-  }
+// function buildObjectDeepSorted(targetValue: any) {
```

Comment:
This is the implementation from earlier today. I modified and rearranged it (see below) to simplify the logic somewhat and added generic type handling which overall has reduced the number of errors, but there is still one left.